fractal_server/app/routes/api/v2/workflow_import.py

```diff
@@ -0,0 +1,230 @@
+                        LinkUserGroup.user_id == user.id
+                    )
+                ),
+            )
+        )
+    )
+    res = await db.execute(stm)
+    task = res.scalars().all()
+    if len(task) > 1:
+        return await _resolve_task_conflict(task, user, db, default_group_id)
+    elif len(task) == 1:
+        return task[0]
+    else:
+        return None
+
+
+async def _get_latest_user_task(task_user_list, db, wf_task) -> TaskV2:
+    """Retrieve the latest task for the user."""
+    if not task_user_list:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Expected one TaskV2 to match the imported one.",
+        )
+    latest_task = max(task_user_list, key=lambda t: t.version)
+
+    stm = (
+        select(TaskV2)
+        .join(TaskGroupV2)
+        .where(TaskGroupV2.id == TaskV2.taskgroupv2_id)
+        .where(TaskGroupV2.pkg_name == wf_task.task.pkg_name)
+        .where(TaskV2.name == wf_task.task.name)
+        .where(TaskGroupV2.version == latest_task.version)
+    )
+    res = await db.execute(stm)
+    task = res.scalars().one_or_none()
+    return task
+
+
+async def _resolve_task_conflict(
+    task_list, user, db, default_group_id
+) -> TaskV2:
+    """Resolve conflicts when multiple tasks match."""
+    for task in task_list:
+        if task.taskgroupv2_id == default_group_id:
+            return task
+    task = task_list[0]
+    stm = (
+        select(TaskV2)
+        .join(TaskGroupV2)
+        .join(LinkUserGroup)
+        .where(LinkUserGroup.group_id == TaskGroupV2.user_group_id)
+        .where(TaskGroupV2.id == TaskV2.taskgroupv2_id)
+        .where(TaskV2.name == task.name)
+        .where(TaskGroupV2.user_group_id == task.user_group_id)
+        .where(LinkUserGroup.user_id == user.id)
+        .order_by(LinkUserGroup.timestamp_created.asc().limit(1))
+    )
+    res = await db.execute(stm)
+    task = res.scalars().one_or_none()
+    return task
+
+
+@router.post(
+    "/project/{project_id}/workflow/import/",
+    response_model=WorkflowReadV2,
+    status_code=status.HTTP_201_CREATED,
+)
+async def import_workflow(
+    project_id: int,
+    workflow: WorkflowImportV2,
+    user: UserOAuth = Depends(current_active_user),
+    db: AsyncSession = Depends(get_async_db),
+) -> WorkflowReadV2:
+    """
+    Import an existing workflow into a project and create required objects.
+    """
+
+    # Preliminary checks
+    await _get_project_check_owner(
+        project_id=project_id, user_id=user.id, db=db
+    )
+    await _check_workflow_exists(
+        name=workflow.name, project_id=project_id, db=db
+    )
+
+    # Create new Workflow
+    db_workflow = WorkflowV2(
+        project_id=project_id,
+        **workflow.dict(exclude_none=True, exclude={"task_list"})
+    )
+    db.add(db_workflow)
+    await db.commit()
+    await db.refresh(db_workflow)
+
+    task_user_list = await _get_user_tasks(user.id, db)
```

Comment:
Rename into `_get_accessible_tasks`, which performs an `or` query (either "my" tasks or the ones of my groups).